src/Promitor.Core.Scraping/Configuration/Serialization/v1/Core/AzureMetricConfigurationDeserializer.cs

```diff
@@ -7,18 +7,43 @@ namespace Promitor.Core.Scraping.Configuration.Serialization.v1.Core
 {
     public class AzureMetricConfigurationDeserializer : Deserializer<AzureMetricConfigurationV1>
     {
+        private const string MultipleDimensionsTag = "dimensions";
+        private const string SingleDimensionTag = "dimension";
+        private readonly IDeserializer<MetricDimensionV1> _dimensionDeserializer;
+            
         public AzureMetricConfigurationDeserializer(IDeserializer<MetricDimensionV1> dimensionDeserializer, IDeserializer<MetricAggregationV1> aggregationDeserializer, ILogger<AzureMetricConfigurationDeserializer> logger)
             : base(logger)
         {
             Map(config => config.MetricName)
                 .IsRequired();
             Map(config => config.Limit)
                 .MapUsing(DetermineLimit);
-            Map(config => config.Dimension)
-                .MapUsingDeserializer(dimensionDeserializer);
             Map(config => config.Aggregation)
                 .IsRequired()
                 .MapUsingDeserializer(aggregationDeserializer);
+            IgnoreField(MultipleDimensionsTag);
+            IgnoreField(SingleDimensionTag);
```

Comment:
I implemented this analogous to how the `resources` property is handled in `MetricDefinitionDeserializer`, first just to handle the list of dimensions correctly, but then also to add backwards compatibility.
The deserialization of those two properties happens in the `Deserialize` method below.

Reply:
Just an idea - But wouldn't it be simpler to keep both fields in the .NET POCO, deserialize both and do the consolidation afterwards? Once the old field goes way, we can just delete the field which also feels more clear when reading the code (instead of having to know this magic is located somewhere)

Reply:
I’ve added/fixed support for YamlSequenceNodes in the Deserializer now, which means the manual `_dimensionDeserializer.Deserialize((YamlSequenceNode)multipleDimensionsNode, errorReporter);` call is not necessary anymore, so the deserializer can be as simple as before again.
(I’m not 100% sure if this change is okay to do, since I assume there was a reason for the old setup, but as far as I understand it still behaves the same as before for YamlMappingNodes, including working Unit tests)

I’ve also re-added the Dimension property and moved the consolidation to `DetermineMetricDimensions`. I am not entirely convinced by that location, but don’t know where else it could be added more nicely. I thought about adding a custom AutoMapper config, but then it would be somewhat hidden again, and we could not check whether both are set in the AzureMetricConfigurationValidator anymore :thinking: